src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -73,10 +73,14 @@ private[health] class HealthCheckActor(
     }
   }
 
+  def getActiveTaskForInstances(instances: Seq[Instance]): Set[Task.Id] = {
+    instances.filter(_.tasksMap.size != 0).map(_.appTask).filter(_.isActive).map(_.taskId)(collection.breakOut)
+  }
+
   def purgeStatusOfDoneInstances(instances: Seq[Instance]): Unit = {
     logger.debug(s"Purging health status of inactive instances for app ${app.id} version ${app.version} and healthCheck ${healthCheck}")
 
-    val activeTaskIds: Set[Task.Id] = instances.map(_.appTask).filter(_.isActive).map(_.taskId)(collection.breakOut)
+    val activeTaskIds: Set[Task.Id] = getActiveTaskForInstances(instances)
```

Comment:
There are other calls to `Instance.appTask`, see L117 and L150.

Reply:
The require in L116 ensure there is one task so L117 should not be a problem there.

Reply:
Should I add a similar `require` in `checkEnoughInstancesRunning` to protect L150 or just move the L116 to L111 just before the call to `checkEnoughInstancesRunning` ?

Reply:
Sounds reasonable, let's try this